solutions/usaco-1088.mdx

```diff
@@ -7,7 +7,187 @@ author: Melody Yu (Video), Tyler Xiao (Video), Maggie Liu
 
 [Official Analysis](http://usaco.org/current/data/sol_prob3_silver_jan21.html)
 
-## Video Solution
+## Internal Video Solution
+
+By Maggie Liu
+
+<Youtube id="u6Qlpo9qgvg" />
+
+<Spoiler title="Video Solution Code">
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>
+using namespace std;
+
+int main()
+{
+	int n;
+	cin >> n;
+	int grid[1000][1000];
+	for (int i = 0; i < n; i++)
+	{
+		for (int j = 0; j < n; j++)
+		{
+			cin >> grid[i][j];
+		}
+	}
+
+	int rows_alternate = 0, cols_alternate = 0;
+	// each row has an alternating pattern of cow and no cow
+	for (int i = 0; i < n; i++)
+	{
+		int sum[2]{};
+		// there are two ways to alternate, odd index and even index
+		for (int j = 0; j < n; j++)
+		{
+			sum[j % 2] += grid[i][j]; // j % 2 is 0 for even j and 1 for odd j
+		}
+		// add the best way to the answer for alternating rows
+		rows_alternate += max(sum[0], sum[1]);
+	}
+
+	// each column has an alternating pattern of cow and no cow
+	for (int i = 0; i < n; i++)
+	{
+		int sum[2]{};
+		// two ways to alternate
+		for (int j = 0; j < n; j++)
+		{
+			sum[j % 2] += grid[j][i];
+		}
+		// add the best way to the answer for alternating columns
+		cols_alternate += max(sum[0], sum[1]);
+	}
+
+	cout << max(rows_alternate, cols_alternate) << endl;
+	return 0;
+}
+
+```
+
+</CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class SpacedOut
+{
+	public static void main(String[] args) throws IOException
+	{
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int[][] grid = new int[n][n];
+		for (int i = 0; i < n; i++)
+		{
+			for (int j = 0; j < n; j++)
+			{
+				grid[i][j] = io.nextInt();
+			}
+		}
+
+		int rowsAlternate = 0, colsAlternate = 0;
+		// each row has an alternating pattern of cow and no cow
+		for (int i = 0; i < n; i++)
+		{
+			int[] sum = new int[2];
+			// there are two ways to alternate, odd index and even index
+			for (int j = 0; j < n; j++)
+			{
+				sum[j % 2] += grid[i][j]; // j % 2 is 0 for even j and 1 for odd j
+			}
+			// add the best way to the answer for alternating rows
+			rowsAlternate += Math.max(sum[0], sum[1]);
+		}
+
+		// each column has an alternating pattern of cow and no cow
+		for (int i = 0; i < n; i++)
+		{
+			int[] sum = new int[2];
+			// two ways to alternate
+			for (int j = 0; j < n; j++)
+			{
+				sum[j % 2] += grid[j][i];
+			}
+			// add the best way to the answer for alternating columns
+			colsAlternate += Math.max(sum[0], sum[1]);
+		}
+
+		io.println(Math.max(rowsAlternate, colsAlternate));
+		io.close();
+	}
+
+	// https://usaco.guide/general/io?lang=java#io-template
+	static class Kattio extends PrintWriter {
```

Comment:
use a codesnip

Reply:
you can replace this entire thing with just
`//CodeSnip{Kattio}`

Reply:
CRAP